frontend/src/pages/journal/components/delete-journal-modal/delete-journal-modal.tsx

```diff
@@ -0,0 +1,67 @@
+import { forwardRef } from 'react';
+
+import { Button, Modal } from '#libs/components/components.js';
+import { DataStatus } from '#libs/enums/enums.js';
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+} from '#libs/hooks/hooks.js';
+import { actions as journalActions } from '#slices/journal/journal.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  id: number | null;
+};
+
+const DeleteJournalModal: React.ForwardRefRenderFunction<
+  HTMLDialogElement,
+  Properties
+> = ({ id }, reference) => {
+  const journalDeleteModalReference =
+    reference as React.RefObject<HTMLDialogElement | null>;
+  const dispatch = useAppDispatch();
+  const handleCancel = useCallback(() => {
+    journalDeleteModalReference.current?.close();
+  }, [journalDeleteModalReference]);
+
+  const { isLoading } = useAppSelector(({ journal }) => {
+    return {
+      isLoading: journal.deleteJournalEntryDataStatus === DataStatus.PENDING,
+    };
+  });
+
+  const handleDeleteJournalEntry = useCallback(() => {
+    void dispatch(journalActions.deleteJournal(id as number)).finally(() => {
+      journalDeleteModalReference.current?.close();
+    });
```

Comment:
```suggestion
    await dispatch(journalActions.deleteJournal(id as number))
     journalDeleteModalReference.current?.close();
```

Reply:
@Liza-Veis if i do like this i have to make this useCallback async, so this cause to changing type of fucntion onClick in button. So should i do that you have written or leave it?

Reply:
yes, make it async

Reply:
done